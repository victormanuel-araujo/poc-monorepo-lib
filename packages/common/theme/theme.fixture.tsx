import React from 'react'

// import { render } from '@testing-library/react-native'
// import { ThemeProvider } from 'styled-components/native'

// import { createTheme } from './'

// export function ThemeFixture({ children }) {
//   return <ThemeProvider theme={createTheme()}>{children}</ThemeProvider>
// }

// export function renderWithTheme(element) {
//   return render(<ThemeFixture>{element}</ThemeFixture>)
// }
