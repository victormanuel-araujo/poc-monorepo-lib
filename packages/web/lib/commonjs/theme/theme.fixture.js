import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createTheme } from "./index";
export function ThemeFixture({
  children
}) {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: createTheme()
  }, children);
}
export function renderWithTheme(element) {
  return render( /*#__PURE__*/React.createElement(ThemeFixture, null, element));
}
//# sourceMappingURL=theme.fixture.js.map