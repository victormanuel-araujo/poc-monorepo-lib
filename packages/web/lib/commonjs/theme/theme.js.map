{"version":3,"sources":["theme.ts"],"names":["colors","fonts","fontSizes","fontWeights","lineHeights","paddings","units","theme"],"mappings":"AACA,SAASA,MAAT,EAAiBC,KAAjB,EAAwBC,SAAxB,EAAmCC,WAAnC,EAAgDC,WAAhD,EAA6DC,QAA7D,EAAuEC,KAAvE;AACA;AACA;AACA;AACA;;AAEA;AACA;AACA;;AACA,OAAO,MAAMC,KAAY,GAAG;AAC1BP,EAAAA,MAD0B;AAE1BC,EAAAA,KAF0B;AAG1BC,EAAAA,SAH0B;AAI1BC,EAAAA,WAJ0B;AAK1BC,EAAAA,WAL0B;AAM1BE,EAAAA,KAN0B;AAO1BD,EAAAA;AAP0B,CAArB","sourcesContent":["import { Theme } from './theme.types'\nimport { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from './theme.values'\n/*\n * in the future, if exists a second theme, just have to create a second\n * composition as same is `theme`.\n */\n\n/**\n * Default gama theme - SMASH\n */\nexport const theme: Theme = {\n  colors,\n  fonts,\n  fontSizes,\n  fontWeights,\n  lineHeights,\n  units,\n  paddings,\n}\n"]}