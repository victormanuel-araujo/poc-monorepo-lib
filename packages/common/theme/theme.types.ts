import { ThemedStyledProps } from 'styled-components'

import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from './theme.values'
export { ThemeProps } from 'styled-components/native'

// definitions of theme types

/**
 * the name of the colors of theme
 */
export type ThemeColorsName = 'primary' | 'secondary' | 'black' | 'white' | 'gray' | 'red' | 'transparent' | 'yellow'

/**
 * Existing tones for each color (ThemeColorsName). Each one color must have the
 * tone "base" and optionally the "light", "lighter", "dark" and "darker" tones
 */
export interface ThemeColorsShade {
  base: string
  light?: string
  lighter?: string
  dark?: string
  darker?: string
}

/**
 * The group of the colors (ThemeColorsName) and shades (ThemeColorsShade)
 *
 * @example
 * Each one color in ThemeColorsName becomes a key of an object, this object must have
 * the shades of ThemeColorsShade
 * ```json
 * {
 *   primary: {
 *     base: '#fff',
 *     light: '#fff',
 *     lighter: '#fff',
 *   }
 * }
 * ```
 */
export type ThemeColorsGroup = { [color in ThemeColorsName]: ThemeColorsShade }

/**
 * Final type of theme colors, bases on colors defined before.
 * that is: Union of ThemeColorsName and ThemeColorsShades with their values defined
 */
export type ThemeColors = typeof colors

/**
 * Type created based on values of font families defined on theme creation.
 * That is: If a family needs to be adde or removed a family, the type are
 * updated automatically
 */
export type ThemeFonts = { [fontName in keyof typeof fonts]: string }
/**
 * Type created based on font sizes also defined when creating the theme
 * @example
 * Sizes defined as follows:
 * ```json
 * {
 *   sm: '0px' as '0px',
 *   md: '0px' as '0px',
 *   ...
 * }
 * ```
 */
export type ThemeFontSizes = { [fontSize in keyof typeof fontSizes]: number }

/**
 * Type of font weights, based on values defined also in the theme creation
 * @example
 * Weights defined as follows:
 * ```json
 * {
 *   regular: 500 as 500,,
 *   bold: 900 as 900,
 *   ...
 * }
 * ```
 */
export type ThemeFontWeights = { [fontSize in keyof typeof fontWeights]: number }

/**
 * Type of the line heights, based on their values defined
 */
export type ThemeLineHeights = { [fontSize in keyof typeof lineHeights]: number }

/**
 * Type created based on measure units values defineds on theme creation
 * @example
 * Any units as example:
 * - base unit of spacing: 8
 * - header height (mobile): 60
 */
export type ThemeUnits = { [fontSize in keyof typeof units]: number }

/**
 * Type of pre-definitions of paddings (using base unit). Pre-definition also created
 * in initial theme values.
 * @example
 * The padding values are defined using the base unit, that is:
 * ```json
 * {
 *   sm: units.base as 8,
 *   md: (units.base * 2) as 16,
 *   lg: (units.base * 3) as 24,
 *   ...
 * }
 * ```
 */
export type ThemePaddings = { [fontSize in keyof typeof paddings]: number }

/**
 * The final interface of the theme.
 * The union of all types/interfaces created before.
 */
export interface Theme {
  colors: ThemeColors

  fonts: ThemeFonts
  fontSizes: ThemeFontSizes
  fontWeights: ThemeFontWeights
  lineHeights: ThemeLineHeights

  units: ThemeUnits
  paddings: ThemePaddings
}

/**
 * Inject the theme props into a styled component.
 */
export type ThemedProps<P> = ThemedStyledProps<P, Theme>

declare module 'styled-components/native' {
  export interface DefaultTheme extends Theme {}
}
