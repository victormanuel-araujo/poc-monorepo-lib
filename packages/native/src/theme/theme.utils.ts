// utils functions to handle theme
import { get, defaultsDeep } from 'lodash'
import { Platform } from 'react-native'
import { css } from 'styled-components/native'

import { theme as _theme } from './theme'
import {
  Theme,
  ThemeColorsName,
  ThemeColorsShade,
  ThemeFontWeights,
  ThemeFontSizes,
  ThemePaddings,
  ThemeProps,
  ThemeFonts,
} from './theme.types'

/**
 * This function create an theme object.
 * Optionally, you can pass an custom theme object with custom values and it will
 * always be merged with the default theme (smash)
 *
 * @param theme - custom theme
 *
 * @example
 * ```ts
 * const theme_a = { a: 1, b: 2, c: 3 };
 * const theme_b = { a: 2 };
 *
 * const theme_c = createTheme(theme_a, theme_b);
 *
 * // the output (theme_c) is equal to:
 * // theme_c = { a: 2, b: 2, c: 3 };
 * ```
 */
export const createTheme = (theme?: DeepPartial<Theme>): Theme => {
  return defaultsDeep(theme, _theme)
}

/**
 * easy way to get an theme color with a shade and optionally an alpha channel
 *
 * @param colorName - the name of the color
 * @param colorShade - the shade of the color
 * @param alpha - an alpha value in hexadecimal pattern, that is: two character from '00' to 'FF'
 *
 * @example
 * ```ts
 * inside a styled-component
 * ${getThemeColor(...props)}
 *
 * outside a styled-component
 * getThemeColor(...props)({ theme })
 * ```
 */
export const getThemeColor =
  (colorName: ThemeColorsName, colorShade: keyof ThemeColorsShade = 'base', alpha?: string) =>
  ({ theme }): string => {
    const color = get(theme, ['colors', colorName], { base: '#000000' })
    const shade: string = get(color, [colorShade], color.base)

    return alpha && alpha.length === 2 ? `${shade}${alpha}` : shade
  }

/**
 * Get the style of a font weight.
 * Verify the platform where the component is used, and return te correct style
 * to apply the font weight
 *
 * @param weight - Font weight, ie light, regular, bold, etc.
 * @param family - Font family - only used on Android
 *
 * @example
 * ```ts
 * const Bold = styled.Text`
 *   ${getFontWeight('bold')}
 * `
 * ```
 */
export const getFontWeight =
  (weight: keyof ThemeFontWeights, family: keyof ThemeFonts = 'default') =>
  ({ theme }) => {
    /*
     * In android, is necessary to change the font family too, cause the
     * font-weight property not work correctly
     */
    if (Platform.OS === 'android') {
      /* get the font family based on "family" prop  */
      const fontFamily: string = get(theme, ['fonts', family])

      /* return the style, with font-family and font-weight defined correctly */
      return css`
        font-family: ${fontFamily}-${weight};
        font-weight: ${get(theme, ['fontWeights', weight])};
      `
    }

    /*
     * In another platforms, as IOS or Web, is not necessary to set a new font family
     * so,
     */
    return css`
      font-weight: ${theme.fontWeights[weight]};
    `
  }

/**
 * Get the correct style for a font size.
 * return a style, with the font size and the relative line-height.
 *
 * @param size - Font size name, ie sm, md, lg, etc.
 * @param rawValue - If it's true, returns the raw integer value
 *                   instead of a style.
 *
 * @example
 * ```ts
 * const Title = styled.Text`
 *   ${getFontSize('lg')};
 * `
 * ```
 *
 * @example - `theme` comes from styled-component's ThemeContext.
 * ```tsx
 *   return <Icon size={getFontSize('lg', true)({ theme })} />
 * ```
 */
export const getFontSize =
  (size: keyof ThemeFontSizes, rawValue = false) =>
  ({ theme }: ThemeProps<Theme>) => {
    if (rawValue) {
      return get(theme, ['fontSizes', size])
    }

    return css`
      font-size: ${get(theme, ['fontSizes', size])}px;
      line-height: ${get(theme, ['lineHeights', size])}px;
    `
  }

/**
 * Used to get a style of a font family and verify the platform where the
 * component is used, and return te correct style
 * to apply the font family
 *
 * @param family - the name of the family
 * @param weight - optionally an weight of the family. use it only in android
 *
 * *
 * @example
 * ```ts
 * const TitleFamily = styled.Text`
 *   ${getFontFamily('title')}
 * `
 * ```
 */
export const getFontFamily =
  (family: keyof ThemeFonts = 'default', weight: keyof ThemeFontWeights = 'medium') =>
  ({ theme }: ThemeProps<Theme>) => {
    /* get the font family based on "family" prop  */
    const fontFamily: string = get(theme, ['fonts', family])

    if (Platform.OS === 'android') {
      /* return the style, with font-family defined correctly */
      return css`
        font-family: ${fontFamily}-${weight};
      `
    }

    return css`
      font-family: ${fontFamily};
    `
  }

/**
 * Get a named padding from current theme.
 *
 * @param size - Padding size name, ie sm, md, lg, etc.
 * @param multiply - used if is necessary to multiply the padding value
 *
 * @example
 * ```ts
 * const Container = styled.View`
 *   padding: ${getPadding('md')} ${getPadding('lg')};
 * `
 * ```
 */
export const getPadding =
  (size: keyof ThemePaddings, multiply = 1) =>
  ({ theme }: ThemeProps<Theme>): string => {
    return theme.paddings[size] * multiply + 'px'
  }
