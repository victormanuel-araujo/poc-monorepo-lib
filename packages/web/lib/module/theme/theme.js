import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from "./theme.values";
/*
 * in the future, if exists a second theme, just have to create a second
 * composition as same is `theme`.
 */

/**
 * Default gama theme - SMASH
 */

export const theme = {
  colors,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  units,
  paddings
};
//# sourceMappingURL=theme.js.map