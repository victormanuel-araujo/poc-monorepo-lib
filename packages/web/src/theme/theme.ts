import { Theme } from './theme.types'
import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from './theme.values'
/*
 * in the future, if exists a second theme, just have to create a second
 * composition as same is `theme`.
 */

/**
 * Default gama theme - SMASH
 */
export const theme: Theme = {
  colors,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  units,
  paddings,
}
