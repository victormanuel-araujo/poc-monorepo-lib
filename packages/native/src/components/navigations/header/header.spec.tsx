import React from 'react'

import { Header, Icon } from '@components'
import { fireEvent } from '@testing-library/react-native'

import { renderWithTheme } from '../../../theme/theme.fixture'

describe('Header Component', () => {
  it('should render correctly', () => {
    const component = renderWithTheme(<Header />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render the right adornment correctly', () => {
    const component = renderWithTheme(<Header rightAdornment={<Icon.Material icon="close" size={20} shape="none" />} />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render the left adornment correctly', () => {
    const component = renderWithTheme(<Header leftAdornment={<Icon.Material icon="close" size={20} shape="none" />} />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should to call the right action function', () => {
    const fn = jest.fn()
    const component = renderWithTheme(
      <Header rightAction={fn} rightAdornment={<Icon.Material icon="close" size={20} shape="none" />} />
    )

    const button = component.queryByTestId('right-action')
    fireEvent.press(button)

    expect(fn).toHaveBeenCalled()
  })

  it('should to call the left action function', () => {
    const fn = jest.fn()
    const component = renderWithTheme(
      <Header leftAction={fn} leftAdornment={<Icon.Material icon="arrow-back" size={20} shape="none" />} />
    )

    const button = component.queryByTestId('left-action')
    fireEvent.press(button)

    expect(fn).toHaveBeenCalled()
  })

  it('should to render the right and left adornment correctly with their props', () => {
    const component = renderWithTheme(
      <Header
        leftAdornment={<Icon.Material icon="arrow-back" size={20} shape="none" />}
        leftAdornmentProps={{ spacing: 'xl' }}
        rightAdornment={<Icon.Material icon="close" size={20} shape="none" />}
        rightAdornmentProps={{ spacing: 'sm', background: 'red' }}
      />
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should to render the title correctly with it props', () => {
    const component = renderWithTheme(<Header title="Lorem Ipsum" titleProps={{ color: 'red', size: 'sm' }} />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
