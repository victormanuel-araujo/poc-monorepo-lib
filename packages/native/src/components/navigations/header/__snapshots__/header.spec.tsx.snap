// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
<View
  accessibilityRole="header"
  backgroundColor="white"
  backgroundShade="base"
  dir="column"
  display="flex"
  position="relative"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minHeight": 60,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="left-action"
    />
  </View>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    flex={1}
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    testID="view"
  >
    <Text
      accessibilityRole="text"
      align="center"
      color="black"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Raleway",
          "fontSize": 16,
          "fontWeight": "700",
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      type="description"
      weight="bold"
    />
  </View>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="right-action"
    />
  </View>
</View>
`;

exports[`Header Component should render the left adornment correctly 1`] = `
<View
  accessibilityRole="header"
  backgroundColor="white"
  backgroundShade="base"
  dir="column"
  display="flex"
  position="relative"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minHeight": 60,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="left-action"
    >
      <View
        shape="none"
        size={20}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 28,
              "justifyContent": "center",
              "width": 28,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          Óóç
        </Text>
      </View>
    </View>
  </View>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    flex={1}
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    testID="view"
  >
    <Text
      accessibilityRole="text"
      align="center"
      color="black"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Raleway",
          "fontSize": 16,
          "fontWeight": "700",
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      type="description"
      weight="bold"
    />
  </View>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="right-action"
    />
  </View>
</View>
`;

exports[`Header Component should render the right adornment correctly 1`] = `
<View
  accessibilityRole="header"
  backgroundColor="white"
  backgroundShade="base"
  dir="column"
  display="flex"
  position="relative"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minHeight": 60,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="left-action"
    />
  </View>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    flex={1}
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    testID="view"
  >
    <Text
      accessibilityRole="text"
      align="center"
      color="black"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Raleway",
          "fontSize": 16,
          "fontWeight": "700",
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      type="description"
      weight="bold"
    />
  </View>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="right-action"
    >
      <View
        shape="none"
        size={20}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 28,
              "justifyContent": "center",
              "width": 28,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          Óóç
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Header Component should to render the right and left adornment correctly with their props 1`] = `
<View
  accessibilityRole="header"
  backgroundColor="white"
  backgroundShade="base"
  dir="column"
  display="flex"
  position="relative"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minHeight": 60,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    spacing="xl"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 32,
        "paddingLeft": 32,
        "paddingRight": 32,
        "paddingTop": 32,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="left-action"
    >
      <View
        shape="none"
        size={20}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 28,
              "justifyContent": "center",
              "width": 28,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ÓóÑ
        </Text>
      </View>
    </View>
  </View>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    flex={1}
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    testID="view"
  >
    <Text
      accessibilityRole="text"
      align="center"
      color="black"
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Raleway",
          "fontSize": 16,
          "fontWeight": "700",
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
      type="description"
      weight="bold"
    />
  </View>
  <View
    background="red"
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    spacing="sm"
    style={
      Object {
        "backgroundColor": "#DB4446",
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="right-action"
    >
      <View
        shape="none"
        size={20}
        style={
          Array [
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 28,
              "justifyContent": "center",
              "width": 28,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          Óóç
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Header Component should to render the title correctly with it props 1`] = `
<View
  accessibilityRole="header"
  backgroundColor="white"
  backgroundShade="base"
  dir="column"
  display="flex"
  position="relative"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minHeight": 60,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="left-action"
    />
  </View>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    flex={1}
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    testID="view"
  >
    <Text
      accessibilityRole="text"
      align="center"
      color="red"
      ellipsizeMode="tail"
      numberOfLines={1}
      size="sm"
      style={
        Object {
          "color": "#DB4446",
          "fontFamily": "Raleway",
          "fontSize": 12,
          "fontWeight": "700",
          "lineHeight": 18,
          "textAlign": "center",
        }
      }
      type="description"
      weight="bold"
    >
      Lorem Ipsum
    </Text>
  </View>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": 60,
        "minWidth": 50,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 60,
          "justifyContent": "center",
          "minWidth": 50,
          "opacity": 1,
          "width": "auto",
        }
      }
      testID="right-action"
    />
  </View>
</View>
`;
